azure-pipelines.yml

```diff
@@ -100,11 +106,11 @@ jobs:
       - task: PublishTestResults@2
         inputs:
           testRunner: JUnit
-          testResultsFiles: $(sdkPath)/report.xml
+          testResultsFiles: $(sdkPath)/**/report.xml
           failTaskOnFailedTests: true
 
       - task: PublishCodeCoverageResults@1
         inputs:
           codeCoverageTool: Cobertura 
           summaryFileLocation: $(sdkPath)/coverage.xml
-          additionalCodeCoverageFiles: $(sdkPath)/coverage.html
+          additionalCodeCoverageFiles: $(sdkPath)/**/coverage.html
```

Comment:
Not sure if this one will work as expected; each module will now produce an `xml` and a `html`, but this is only uploading the `html`, and a `coverage.xml` of the "main" module (which doesn't exist, so nothing will be stored there)

Perhaps you're able to check if the uploaded results produce something useful now.

Reply:
I made a slight change to merge all the coverage.txt files together (we do the same thing in azure-sdk-for-go).

Reply:
Ah thanks! It was getting late here, and I didn't dig yet to see if that was possible, so thanks for that!

Reply:
arf 😢 looks like it's not happy with that change;

```
error: can't find "config.go": cannot find module providing package github.com/Azure/go-autorest/autorest/adal: working directory is not part of a module
##[error]Bash exited with code '1'.
Finishing: Run Tests
```

Reply:
I guess because the root `/` is not a module, so those tools won't work outside of a module 😞 

Perhaps doing a merge within the loop could work (using `$(sdkPath)` for the target directory to write to

Reply:
Yes, apparently `gocov` is required to be run from the module directory.

Reply:
Let me give that a quick try

Reply:
I wonder if it can be _any_ module directory or if it will have to be per each, which then defeats joining them together.

Reply:
Hmm no code coverage info now.  I'm gonna try something, one sec.

Reply:
Ah, yes, I'm stupid. It just pipes the output, and doesn't take the input of "what's already there". Let me fix (hoping it will accept that).

Reply:
If that doesn't work, I'm wondering if `gocovmerge` works with `GO111MODULE=off` (or `auto`); were go modules required to make it work?

Because if it can work without, we could take your approach, but then keep go modules disabled for that step 🤔

Reply:
My idea was to merge after the loop, then `cd autorest` and run `gocov` from there.

Reply:
So, the good new is "CI is still green" with the current approach, but I don't think I have access to the published results, so I'm not sure the merge worked.

Feel free to amend my last commit to try your approach if it didn't work. I'm curious why it needs a module though; I hope running it from within a non-related module won't cause issues 🤔